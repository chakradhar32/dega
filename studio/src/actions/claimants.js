import axios from 'axios';
import {
  ADD_CLAIMANT,
  ADD_CLAIMANTS,
  ADD_CLAIMANTS_REQUEST,
  SET_CLAIMANTS_LOADING,
  RESET_CLAIMANTS,
  CLAIMANTS_API,
} from '../constants/claimants';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { addMedia } from './media';
import getError from '../utils/getError';

export const getClaimants = (query) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .get(CLAIMANTS_API, {
        params: query,
      })
      .then((response) => {
        dispatch(
          addMedia(
            response.data.nodes
              .filter((claimant) => claimant.medium)
              .map((claimant) => claimant.medium),
          ),
        );
        dispatch(
          addClaimantsList(
            response.data.nodes.map((claimant) => {
              return { ...claimant, medium: claimant.medium?.id };
            }),
          ),
        );
        dispatch(
          addClaimantsRequest({
            data: response.data.nodes.map((item) => item.id),
            query: query,
            total: response.data.total,
          }),
        );
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const getClaimant = (id) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .get(CLAIMANTS_API + '/' + id)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(getClaimantByID({ ...response.data, medium: response.data.medium?.id }));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const addClaimant = (data) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .post(CLAIMANTS_API, data)
      .then(() => {
        dispatch(resetClaimants());
        dispatch(addSuccessNotification('Claimant added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const updateClaimant = (data) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .put(CLAIMANTS_API + '/' + data.id, data)
      .then((response) => {
        if (response.data.medium) dispatch(addMedia([response.data.medium]));

        dispatch(getClaimantByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(addSuccessNotification('Claimant updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      })
      .finally(() => dispatch(stopClaimantsLoading()));
  };
};

export const deleteClaimant = (id) => {
  return (dispatch) => {
    dispatch(loadingClaimants());
    return axios
      .delete(CLAIMANTS_API + '/' + id)
      .then(() => {
        dispatch(resetClaimants());
        dispatch(addSuccessNotification('Claimant deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(getError(error)));
      });
  };
};

export const addClaimants = (claimants) => {
  return (dispatch) => {
    dispatch(
      addMedia(claimants.filter((claimant) => claimant.medium).map((claimant) => claimant.medium)),
    );
    dispatch(
      addClaimantsList(
        claimants.map((claimant) => {
          return { ...claimant, medium: claimant.medium?.id };
        }),
      ),
    );
  };
};

export const loadingClaimants = () => ({
  type: SET_CLAIMANTS_LOADING,
  payload: true,
});

export const stopClaimantsLoading = () => ({
  type: SET_CLAIMANTS_LOADING,
  payload: false,
});

export const getClaimantByID = (data) => ({
  type: ADD_CLAIMANT,
  payload: data,
});

export const addClaimantsList = (data) => ({
  type: ADD_CLAIMANTS,
  payload: data,
});

export const addClaimantsRequest = (data) => ({
  type: ADD_CLAIMANTS_REQUEST,
  payload: data,
});

export const resetClaimants = () => ({
  type: RESET_CLAIMANTS,
});
