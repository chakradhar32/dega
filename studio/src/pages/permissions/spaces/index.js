import React from 'react';
import SpacePermissionList from './components/PermissionList';
import { Space } from 'antd';
import { useSelector } from 'react-redux';

function SpacePermissions() {
  const superOrg = useSelector(({ admin }) => {
    return admin.organisation;
  });

  return (
    <Space direction="vertical">
      {/* <Link key="1" to="/permissions/spaces/create">
        <Button>Create New</Button>
      </Link> */}
      <SpacePermissionList admin={superOrg.is_admin ? superOrg.is_admin : false} />
    </Space>
  );
}

export default SpacePermissions;
