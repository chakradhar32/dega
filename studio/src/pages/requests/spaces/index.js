import React from 'react';
import SpaceRequestList from './components/RequestList';
import { Space, Button, Row } from 'antd';
import { Link } from 'react-router-dom';

function SpaceRequests() {
  return (
    <Space direction="vertical">
      <Row gutter={16} justify="end">
        <Link key="1" to="/admin/requests/spaces/create">
          <Button type="primary">New Space Request</Button>
        </Link>
      </Row>

      <SpaceRequestList />
    </Space>
  );
}

export default SpaceRequests;
