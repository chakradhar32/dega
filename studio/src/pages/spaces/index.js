import React from 'react';
import { Space, Button, Row, Col } from 'antd';
import { Link } from 'react-router-dom';
import SpaceList from './components/SpaceList';
import { useSelector } from 'react-redux';

function Spaces() {
  const { role } = useSelector((state) => {
    const { selected, orgs } = state.spaces;

    if (selected > 0) {
      const space = state.spaces.details[selected];
      const orgId = space.organisation_id;
      const org = orgs.find((org) => org.id === orgId);
      const role = org.permission.role;
      return {
        role: role,
      };
    }
    return { role: 'member' };
  });

  return (
    <Space direction="vertical">
      <Row gutter={16} justify="end">
        {role === 'owner' ? (
          <Col>
            <Link key="2" to="/reindex">
              <Button>Reindex</Button>
            </Link>
          </Col>
        ) : null}
        <Col>
          <Link key="1" to="/spaces/create">
            <Button type="primary">New Space</Button>
          </Link>
        </Col>
        <Link key="1" to="/admin/spaces/create">
          <Button type="primary">New Space</Button>
        </Link>
      </Row>
      <SpaceList />
    </Space>
  );
}

export default Spaces;
