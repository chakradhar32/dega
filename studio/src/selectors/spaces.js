export const spaceSelector = (state) => {
  const selectedOrg = state.spaces.orgs.find((item) => item.spaces.includes(state.spaces.selected));
  let spaces = [];
  if (selectedOrg) {
    spaces = selectedOrg.spaces.map((s) => state.spaces.details[s]);
  }
  return {
    loading: state.spaces.loading,
    spaces: spaces,
  };
};
