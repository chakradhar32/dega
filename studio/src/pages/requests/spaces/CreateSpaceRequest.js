import React from 'react';
import SpaceRequestCreateForm from './components/RequestForm';
import { useDispatch } from 'react-redux';
import { addSpaceRequest } from '../../../actions/spaceRequests';
import { useHistory } from 'react-router-dom';
import { Helmet } from 'react-helmet';

function CreateSpaceRequest() {
  const history = useHistory();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpaceRequest(values)).then(() => history.push('/admin/requests/spaces'));
  };
  return (
    <>
      <Helmet title={'Create Space Request'} />
      <SpaceRequestCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateSpaceRequest;
