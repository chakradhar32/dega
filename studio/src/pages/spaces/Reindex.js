import React from 'react';
import { Row, Col, Button, Form, Select, Typography, Space } from 'antd';
import { useSelector, useDispatch } from 'react-redux';
import { reindex, reindexSpace } from '../../actions/meiliReindex';

const Reindex = () => {
  const { Title } = Typography;

  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const [spaceOption, setSpaceOption] = React.useState();
  const [status, setStatus] = React.useState();

  const superOrg = useSelector(({ admin }) => {
    return admin.organisation;
  });

  const { orgId, orgs, spaces } = useSelector((state) => {
    const { selected, orgs, loading } = state.spaces;
    if (selected > 0) {
      const space = state.spaces.details[selected];
      const orgId = space.organisation_id;
      const org = orgs.find((org) => org.id === orgId);
      const role = org.permission.role;

      return {
        loading: loading,
        role: role,
        orgId: orgId,
        spaces: state.spaces.details,
        orgs: superOrg.is_admin ? state.spaces.orgs : [org],
      };
    }
    return { orgId: 0, orgs: [], spaces: [] };
  });
  const [organisationId, setOrganisationId] = React.useState(orgId);

  React.useEffect(() => {
    if (orgs.length > 0) {
      const org = orgs.find((org) => org.id === organisationId);
      const newSpaces = org.spaces.map((space) => spaces[space]);
      setSpaceOption(newSpaces);
    }
  }, [organisationId]);

  const handleorgChange = (orgId) => {
    form.setFieldsValue({
      space_id: null,
    });
    setOrganisationId(orgId);
  };
  const handleReindex = () => {
    dispatch(reindex());
  };
  const handleSpaceReindex = (id) => {
    if (id) {
      dispatch(reindexSpace(id));
    } else {
      setStatus('error');
    }
  };
  return (
    <Space direction="vertical">
      <Title level={3}>Reindex</Title>
      <Form form={form} name="reindex" layout="vertical">
        <Row justify="center" style={{ maxWidth: '1200px', width: '100%', margin: '0 auto' }}>
          <Col span={24}>
            <Row
              gutter={40}
              justify="space-around"
              style={{ background: '#f0f2f5', padding: '1.25rem', marginBottom: '1rem' }}
            >
              <Col span={12}>
                {orgs && orgs.length > 0 ? (
                  <Row gutter={40}>
                    <Col md={{ span: 16 }}>
                      <Form.Item name="organisation_id" label="Organisation">
                        <Select
                          allowClear
                          bordered
                          listHeight={128}
                          defaultValue={[orgId]}
                          onChange={(value) => handleorgChange(value)}
                        >
                          {orgs.map((item) => (
                            <Select.Option value={item.id} key={item.id}>
                              {item.title}
                            </Select.Option>
                          ))}
                        </Select>
                      </Form.Item>
                    </Col>
                  </Row>
                ) : null}

                {spaceOption && spaceOption.length > 0 ? (
                  <Row gutter={40}>
                    <Col md={{ span: 16 }}>
                      <Form.Item name="space_id" label="Space" validateStatus={status}>
                        <Select
                          allowClear
                          bordered
                          listHeight={128}
                          defaultValue={[]}
                          placeholder="Select Space"
                          onChange={() => setStatus()}
                        >
                          {spaceOption.map((item) => (
                            <Select.Option value={item.id} key={item.id}>
                              {item.name}
                            </Select.Option>
                          ))}
                        </Select>
                      </Form.Item>
                    </Col>
                    <Col
                      md={{ span: 5 }}
                      style={{ alignItems: 'center', justifyContent: 'center', display: 'flex' }}
                    >
                      <Button
                        type="primary"
                        onClick={() => handleSpaceReindex(form.getFieldValue('space_id'))}
                      >
                        Reindex
                      </Button>
                    </Col>
                  </Row>
                ) : null}
                {superOrg.is_admin ? (
                  <Button type="primary" onClick={() => handleReindex()}>
                    Reindex Instance
                  </Button>
                ) : null}
              </Col>
            </Row>
          </Col>
        </Row>
      </Form>
    </Space>
  );
};

export default Reindex;
