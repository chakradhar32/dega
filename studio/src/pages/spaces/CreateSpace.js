import React from 'react';
import SpaceCreateForm from './components/SpaceCreateForm';
import { useDispatch } from 'react-redux';
import { addSpace, getSpaces } from '../../actions/spaces';
import { useHistory } from 'react-router-dom';
import { Helmet } from 'react-helmet';

function CreateSpace() {
  const history = useHistory();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpace(values)).then(() => {
      dispatch(getSpaces());
      history.push('/admin/spaces');
    });
  };
  return (
    <>
      <Helmet title={'Create Space'} />
      <SpaceCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateSpace;
