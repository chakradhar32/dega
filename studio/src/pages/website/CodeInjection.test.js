import React from 'react';
import { Provider, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { act } from '@testing-library/react';
import { mount } from 'enzyme';

import '../../matchMedia.mock';
import CodeInjection from './CodeInjection';
import { updateSpace } from '../../actions/spaces';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('react-monaco-editor', () => {
  const MonacoEditor = () => <div />;
  return MonacoEditor;
});
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn().mockReturnValue({ id: '11' }),
}));
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('../../actions/spaces', () => ({
  updateSpace: jest.fn(),
}));

let store;

describe('Code Injection Form component', () => {
  store = mockStore({
    spaces: {
      orgs: [{ id: 2, title: 'Organization 2', spaces: [] }],
      details: {
        11: {
          id: 11,
          organisation_id: 1,
          name: 'name',
          slug: 'slug',
          site_title: 'site_title',
          tag_line: 'tag_line',
          description: 'description',
          site_address: 'site_address',
          social_media_urls: {
            facebook: 'facebookUrl',
          },
        },
      },
      loading: false,
      selected: 11,
    },
  });
  useDispatch.mockReturnValue(jest.fn());

  describe('snapshot testing', () => {
    it('should render the component', () => {
      const tree = mount(
        <Provider store={store}>
          <CodeInjection />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should submit form with added data', (done) => {
      updateSpace.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      act(() => {
        wrapper
          .find('FormItem')
          .at(1)
          .find('MonacoEditor')
          .props()
          .onChange({
            target: {
              value: '<html>↵<body>↵<h1>Hi</h1>↵</body>↵</html>',
            },
          });
        wrapper
          .find('FormItem')
          .at(2)
          .find('MonacoEditor')
          .props()
          .onChange({
            target: {
              value: '<html>↵<body>↵<h1>Hi</h1>↵</body>↵</html>',
            },
          });
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Update');
        submitButtom.simulate('submit');
        wrapper.update();
      });

      setTimeout(() => {
        expect(updateSpace).toHaveBeenCalledTimes(1);
        expect(updateSpace).toHaveBeenCalledWith({
          id: 11,
          organisation_id: 1,
          name: 'name',
          slug: 'slug',
          site_title: 'site_title',
          tag_line: 'tag_line',
          description: 'description',
          site_address: 'site_address',
          header_code: '<html>↵<body>↵<h1>Hi</h1>↵</body>↵</html>',
          footer_code: '<html>↵<body>↵<h1>Hi</h1>↵</body>↵</html>',
          social_media_urls: {
            facebook: 'facebookUrl',
          },
        });
        done();
      }, 0);
    });
    it('should display RecordNotFound when no space found', () => {
      store = mockStore({
        spaces: {
          orgs: [],
          details: {},
          loading: false,
          selected: 0,
        },
      });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      expect(wrapper.find('RecordNotFound').length).toBe(1);
    });
    it('should display Skeleton when loading', () => {
      store = mockStore({
        spaces: {
          orgs: [],
          details: {},
          loading: true,
          selected: 0,
        },
      });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      expect(wrapper.find('Skeleton').length).toBe(1);
    });
  });
});
