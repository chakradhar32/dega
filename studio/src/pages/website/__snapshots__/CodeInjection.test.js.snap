// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Injection Form component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CodeInjection>
    <div>
      <ForwardRef(InternalForm)
        form={
          Object {
            "__INTERNAL__": Object {
              "itemRef": [Function],
              "name": "code-injection",
            },
            "getFieldError": [Function],
            "getFieldInstance": [Function],
            "getFieldValue": [Function],
            "getFieldsError": [Function],
            "getFieldsValue": [Function],
            "getInternalHooks": [Function],
            "isFieldTouched": [Function],
            "isFieldValidating": [Function],
            "isFieldsTouched": [Function],
            "isFieldsValidating": [Function],
            "resetFields": [Function],
            "scrollToField": [Function],
            "setFields": [Function],
            "setFieldsValue": [Function],
            "submit": [Function],
            "validateFields": [Function],
          }
        }
        initialValues={
          Object {
            "description": "description",
            "id": 11,
            "name": "name",
            "organisation_id": 1,
            "site_address": "site_address",
            "site_title": "site_title",
            "slug": "slug",
            "social_media_urls": Object {
              "facebook": "facebookUrl",
            },
            "tag_line": "tag_line",
          }
        }
        layout="vertical"
        name="code-injection"
        onFinish={[Function]}
        onFinishFailed={[Function]}
        onValuesChange={[Function]}
        scrollToFirstError={true}
        style={
          Object {
            "paddingTop": "24px",
          }
        }
      >
        <SizeContextProvider>
          <ForwardRef(Form)
            className="ant-form ant-form-vertical"
            form={
              Object {
                "__INTERNAL__": Object {
                  "itemRef": [Function],
                  "name": "code-injection",
                },
                "getFieldError": [Function],
                "getFieldInstance": [Function],
                "getFieldValue": [Function],
                "getFieldsError": [Function],
                "getFieldsValue": [Function],
                "getInternalHooks": [Function],
                "isFieldTouched": [Function],
                "isFieldValidating": [Function],
                "isFieldsTouched": [Function],
                "isFieldsValidating": [Function],
                "resetFields": [Function],
                "scrollToField": [Function],
                "setFields": [Function],
                "setFieldsValue": [Function],
                "submit": [Function],
                "validateFields": [Function],
              }
            }
            id="code-injection"
            initialValues={
              Object {
                "description": "description",
                "id": 11,
                "name": "name",
                "organisation_id": 1,
                "site_address": "site_address",
                "site_title": "site_title",
                "slug": "slug",
                "social_media_urls": Object {
                  "facebook": "facebookUrl",
                },
                "tag_line": "tag_line",
              }
            }
            name="code-injection"
            onFinish={[Function]}
            onFinishFailed={[Function]}
            onValuesChange={[Function]}
            style={
              Object {
                "paddingTop": "24px",
              }
            }
          >
            <form
              className="ant-form ant-form-vertical"
              id="code-injection"
              onReset={[Function]}
              onSubmit={[Function]}
              style={
                Object {
                  "paddingTop": "24px",
                }
              }
            >
              <Row
                justify="end"
              >
                <div
                  className="ant-row ant-row-end"
                  style={Object {}}
                >
                  <FormItem>
                    <Row
                      className="ant-form-item"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item"
                        style={Object {}}
                      >
                        <FormItemLabel
                          prefixCls="ant-form"
                          requiredMark={true}
                        />
                        <FormItemInput
                          errors={Array []}
                          onDomErrorVisibleChange={[Function]}
                          prefixCls="ant-form"
                          status=""
                          validateStatus=""
                        >
                          <Col
                            className="ant-form-item-control"
                          >
                            <div
                              className="ant-col ant-form-item-control"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Button
                                    disabled={true}
                                    htmlType="submit"
                                    type="primary"
                                  >
                                    <Wave>
                                      <button
                                        className="ant-btn ant-btn-primary"
                                        disabled={true}
                                        onClick={[Function]}
                                        type="submit"
                                      >
                                        <LoadingIcon
                                          existIcon={false}
                                          loading={false}
                                          prefixCls="ant-btn"
                                        >
                                          <CSSMotion
                                            motionName="ant-btn-loading-icon-motion"
                                            onAppearActive={[Function]}
                                            onAppearStart={[Function]}
                                            onEnterActive={[Function]}
                                            onEnterStart={[Function]}
                                            onLeaveActive={[Function]}
                                            onLeaveStart={[Function]}
                                            removeOnLeave={true}
                                            visible={false}
                                          >
                                            <DomWrapper />
                                          </CSSMotion>
                                        </LoadingIcon>
                                        <span
                                          key=".0"
                                        >
                                          Update
                                        </span>
                                      </button>
                                    </Wave>
                                  </Button>
                                </div>
                              </div>
                              <ErrorList
                                errors={Array []}
                                onDomErrorVisibleChange={[Function]}
                              >
                                <CSSMotion
                                  motionAppear={true}
                                  motionDeadline={500}
                                  motionName="ant-show-help"
                                  onLeaveEnd={[Function]}
                                  removeOnLeave={true}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </FormItem>
                </div>
              </Row>
              <FormItem
                label="Header Code"
                name="header_code"
              >
                <WrapperField
                  label="Header Code"
                  messageVariables={
                    Object {
                      "label": "Header Code",
                    }
                  }
                  name="header_code"
                  onReset={[Function]}
                  trigger="onChange"
                  validateTrigger="onChange"
                >
                  <Field
                    fieldContext={
                      Object {
                        "__INTERNAL__": Object {
                          "itemRef": [Function],
                          "name": "code-injection",
                        },
                        "getFieldError": [Function],
                        "getFieldInstance": [Function],
                        "getFieldValue": [Function],
                        "getFieldsError": [Function],
                        "getFieldsValue": [Function],
                        "getInternalHooks": [Function],
                        "isFieldTouched": [Function],
                        "isFieldValidating": [Function],
                        "isFieldsTouched": [Function],
                        "isFieldsValidating": [Function],
                        "resetFields": [Function],
                        "scrollToField": [Function],
                        "setFields": [Function],
                        "setFieldsValue": [Function],
                        "submit": [Function],
                        "validateFields": [Function],
                        "validateTrigger": "onChange",
                      }
                    }
                    key="_header_code"
                    label="Header Code"
                    messageVariables={
                      Object {
                        "label": "Header Code",
                      }
                    }
                    name={
                      Array [
                        "header_code",
                      ]
                    }
                    onReset={[Function]}
                    trigger="onChange"
                    validateTrigger="onChange"
                    valuePropName="value"
                  >
                    <Row
                      className="ant-form-item"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item"
                        style={Object {}}
                      >
                        <FormItemLabel
                          htmlFor="code-injection_header_code"
                          label="Header Code"
                          name="header_code"
                          prefixCls="ant-form"
                          required={false}
                          requiredMark={true}
                        >
                          <Col
                            className="ant-form-item-label"
                          >
                            <div
                              className="ant-col ant-form-item-label"
                              style={Object {}}
                            >
                              <label
                                className=""
                                htmlFor="code-injection_header_code"
                                title="Header Code"
                              >
                                Header Code
                              </label>
                            </div>
                          </Col>
                        </FormItemLabel>
                        <FormItemInput
                          errors={Array []}
                          label="Header Code"
                          name={
                            Array [
                              "header_code",
                            ]
                          }
                          onDomErrorVisibleChange={[Function]}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validateStatus=""
                          validating={false}
                        >
                          <Col
                            className="ant-form-item-control"
                          >
                            <div
                              className="ant-col ant-form-item-control"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Component
                                    update={2}
                                  >
                                    <MONACOEditor
                                      id="code-injection_header_code"
                                      language="html"
                                      onChange={[Function]}
                                      width="100%"
                                    >
                                      <MonacoEditor
                                        height="240"
                                        language="html"
                                        onChange={[Function]}
                                        options={
                                          Object {
                                            "autoClosingBrackets": "auto",
                                            "autoClosingQuotes": "auto",
                                            "cursorBlinking": "smooth",
                                            "folding": true,
                                            "lineNumbersMinChars": 4,
                                            "minimap": Object {
                                              "enabled": false,
                                            },
                                            "scrollBeyondLastLine": false,
                                            "theme": "vs-dark",
                                            "wordWrap": "on",
                                          }
                                        }
                                        width="100%"
                                      >
                                        <div />
                                      </MonacoEditor>
                                    </MONACOEditor>
                                  </Component>
                                </div>
                              </div>
                              <ErrorList
                                errors={Array []}
                                onDomErrorVisibleChange={[Function]}
                              >
                                <CSSMotion
                                  motionAppear={true}
                                  motionDeadline={500}
                                  motionName="ant-show-help"
                                  onLeaveEnd={[Function]}
                                  removeOnLeave={true}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </Field>
                </WrapperField>
              </FormItem>
              <FormItem
                label="Footer Code"
                name="footer_code"
              >
                <WrapperField
                  label="Footer Code"
                  messageVariables={
                    Object {
                      "label": "Footer Code",
                    }
                  }
                  name="footer_code"
                  onReset={[Function]}
                  trigger="onChange"
                  validateTrigger="onChange"
                >
                  <Field
                    fieldContext={
                      Object {
                        "__INTERNAL__": Object {
                          "itemRef": [Function],
                          "name": "code-injection",
                        },
                        "getFieldError": [Function],
                        "getFieldInstance": [Function],
                        "getFieldValue": [Function],
                        "getFieldsError": [Function],
                        "getFieldsValue": [Function],
                        "getInternalHooks": [Function],
                        "isFieldTouched": [Function],
                        "isFieldValidating": [Function],
                        "isFieldsTouched": [Function],
                        "isFieldsValidating": [Function],
                        "resetFields": [Function],
                        "scrollToField": [Function],
                        "setFields": [Function],
                        "setFieldsValue": [Function],
                        "submit": [Function],
                        "validateFields": [Function],
                        "validateTrigger": "onChange",
                      }
                    }
                    key="_footer_code"
                    label="Footer Code"
                    messageVariables={
                      Object {
                        "label": "Footer Code",
                      }
                    }
                    name={
                      Array [
                        "footer_code",
                      ]
                    }
                    onReset={[Function]}
                    trigger="onChange"
                    validateTrigger="onChange"
                    valuePropName="value"
                  >
                    <Row
                      className="ant-form-item"
                      key="row"
                    >
                      <div
                        className="ant-row ant-form-item"
                        style={Object {}}
                      >
                        <FormItemLabel
                          htmlFor="code-injection_footer_code"
                          label="Footer Code"
                          name="footer_code"
                          prefixCls="ant-form"
                          required={false}
                          requiredMark={true}
                        >
                          <Col
                            className="ant-form-item-label"
                          >
                            <div
                              className="ant-col ant-form-item-label"
                              style={Object {}}
                            >
                              <label
                                className=""
                                htmlFor="code-injection_footer_code"
                                title="Footer Code"
                              >
                                Footer Code
                              </label>
                            </div>
                          </Col>
                        </FormItemLabel>
                        <FormItemInput
                          errors={Array []}
                          label="Footer Code"
                          name={
                            Array [
                              "footer_code",
                            ]
                          }
                          onDomErrorVisibleChange={[Function]}
                          prefixCls="ant-form"
                          status=""
                          touched={false}
                          validateStatus=""
                          validating={false}
                        >
                          <Col
                            className="ant-form-item-control"
                          >
                            <div
                              className="ant-col ant-form-item-control"
                              style={Object {}}
                            >
                              <div
                                className="ant-form-item-control-input"
                              >
                                <div
                                  className="ant-form-item-control-input-content"
                                >
                                  <Component
                                    update={2}
                                  >
                                    <MONACOEditor
                                      id="code-injection_footer_code"
                                      language="html"
                                      onChange={[Function]}
                                      width="100%"
                                    >
                                      <MonacoEditor
                                        height="240"
                                        language="html"
                                        onChange={[Function]}
                                        options={
                                          Object {
                                            "autoClosingBrackets": "auto",
                                            "autoClosingQuotes": "auto",
                                            "cursorBlinking": "smooth",
                                            "folding": true,
                                            "lineNumbersMinChars": 4,
                                            "minimap": Object {
                                              "enabled": false,
                                            },
                                            "scrollBeyondLastLine": false,
                                            "theme": "vs-dark",
                                            "wordWrap": "on",
                                          }
                                        }
                                        width="100%"
                                      >
                                        <div />
                                      </MonacoEditor>
                                    </MONACOEditor>
                                  </Component>
                                </div>
                              </div>
                              <ErrorList
                                errors={Array []}
                                onDomErrorVisibleChange={[Function]}
                              >
                                <CSSMotion
                                  motionAppear={true}
                                  motionDeadline={500}
                                  motionName="ant-show-help"
                                  onLeaveEnd={[Function]}
                                  removeOnLeave={true}
                                  visible={false}
                                >
                                  <DomWrapper />
                                </CSSMotion>
                              </ErrorList>
                            </div>
                          </Col>
                        </FormItemInput>
                      </div>
                    </Row>
                  </Field>
                </WrapperField>
              </FormItem>
            </form>
          </ForwardRef(Form)>
        </SizeContextProvider>
      </ForwardRef(InternalForm)>
    </div>
  </CodeInjection>
</Provider>
`;
